src/hooks/useReportIDs.tsx

```diff
@@ -0,0 +1,174 @@
+        // Needed for name sorting:
+        reportName: report.reportName,
+        policyName: report.policyName,
+        oldPolicyName: report.oldPolicyName,
+        // Other less obvious properites considered for sorting:
+        ownerAccountID: report.ownerAccountID,
+        currency: report.currency,
+        managerID: report.managerID,
+        // Other important less obivous properties for filtering:
+        parentReportActionID: report.parentReportActionID,
+        parentReportID: report.parentReportID,
+        isDeletedParentAction: report.isDeletedParentAction,
+        isUnreadWithMention: ReportUtils.isUnreadWithMention(report),
+    }) as ChatReportSelector;
+
+const reportActionsSelector = (reportActions: OnyxEntry<OnyxTypes.ReportActions>): ReportActionsSelector =>
+    (reportActions &&
+        Object.values(reportActions).map((reportAction) => {
+            const {reportActionID, actionName, errors = [], originalMessage} = reportAction;
+            const decision = reportAction.message?.[0]?.moderationDecision?.decision;
+
+            return {
+                reportActionID,
+                actionName,
+                errors,
+                message: [
+                    {
+                        moderationDecision: {decision},
+                    },
+                ] as Message[],
+                originalMessage,
+            };
+        })) as ReportActionsSelector;
+
+const policySelector = (policy: OnyxEntry<OnyxTypes.Policy>): PolicySelector =>
+    (policy && {
+        type: policy.type,
+        name: policy.name,
+        avatar: policy.avatar,
+        employeeList: policy.employeeList,
+    }) as PolicySelector;
+
+function ReportIDsContextProvider({
+    children,
+    /**
+     * Only required to make unit tests work, since we
+     * explicitly pass the currentReportID in LHNTestUtils
+     * to SidebarLinksData, so this context doesn't have
+     * access to currentReportID in that case.
+     *
+     * This is a workaround to have currentReportID available in testing environment.
+     */
+    currentReportIDForTests,
+}: ReportIDsContextProviderProps) {
+    const [priorityMode] = useOnyx(ONYXKEYS.NVP_PRIORITY_MODE, {initialValue: CONST.PRIORITY_MODE.DEFAULT});
+    const [chatReports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {selector: chatReportSelector, initialValue: {} as ChatReportSelector});
+    const [policies] = useOnyx(ONYXKEYS.COLLECTION.POLICY, {selector: policySelector, initialValue: {} as PolicySelector});
+    const [allReportActions] = useOnyx(ONYXKEYS.COLLECTION.REPORT_ACTIONS, {selector: reportActionsSelector, initialValue: {} as ReportActionsSelector});
+    const [transactionViolations] = useOnyx(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS, {initialValue: {} as OnyxCollection<OnyxTypes.TransactionViolations>});
+    const [reportsDrafts] = useOnyx(ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT, {initialValue: {} as OnyxCollection<string>});
```

Comment:
```suggestion
    const [chatReports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {selector: chatReportSelector});
    const [policies] = useOnyx(ONYXKEYS.COLLECTION.POLICY, {selector: policySelector});
    const [allReportActions] = useOnyx(ONYXKEYS.COLLECTION.REPORT_ACTIONS, {selector: reportActionsSelector});
    const [transactionViolations] = useOnyx(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS);
    const [reportsDrafts] = useOnyx(ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT);
```

BTW I think we can remove these `initialValue`'s for the collection keys, because if the collection is empty / inexistent both `withOnyx` / `useOnyx` would automatically return `{}`. Thoughts @hayata-suenaga @roryabraham ?

Reply:
I'll defer the answer to @roryabraham as he's familiar with the `useOnyx` hook. But if the `useOnyx` will automatically return `{}`, then I agree that we can remove the `initialValue`.

Reply:
Agree with @fabioh8010 - it also looks like this will lead to significant simplifications because it eliminates the need for the types for `ChatReportSelector`, `PolicySelector`, and `ReportActionsSelector`

Reply:
@roryabraham  I have removed the `initialValues` but we can't remove the dedicated selector types. The reason is that they are used in the selector functions in the same file.